core/metacling/src/TCling.cxx

```diff
@@ -6052,8 +6052,16 @@ static void* LazyFunctionCreatorAutoloadForModule(const std::string& mangled_nam
 
          auto Symbols = RealSoFile->symbols();
          for (auto S : Symbols) {
-            // DO NOT insert to table if symbol was weak or undefined
-            if (S.getFlags() == SymbolRef::SF_Weak || S.getFlags() == SymbolRef::SF_Undefined) continue;
+            uint32_t Flags = S.getFlags();
+            // DO NOT insert to table if symbol was undefined
+            if (Flags & SymbolRef::SF_Undefined)
+               continue;
+
+            // FIXME:
+            // Only insert weak symbols from white-listed libraryes, such as
+            // ROOT libraries.
+            //  if (Flags & SymbolRef::SF_Weak)
+            //     continue;
```

Comment:
I'm confused. Why do you want to load the library when it has the weak symbol? What we want to fix here is the excessive loading of RooFit, no?

Reply:
Some tests fail due to missing `ZN4ROOT8Internal3RDF12RColumnValueIiE5ResetEv`:
```
 nm -A lib/lib*.so | grep 'ZN4ROOT8Internal3RDF12RColumnValueIiE5ResetEv'
lib/libROOTDataFrame.so:00000000002527f4 W _ZN4ROOT8Internal3RDF12RColumnValueIiE5ResetEv
```

Reply:
http://lists.llvm.org/pipermail/llvm-dev/2017-January/108821.html
> "The concept of the weak symbol is that the symbol is marked as a lower priority and can be overridden by another symbol. Only if no other implementation is found will the weak symbol be the one that it used.
> 
> The logical extension of this for the dynamic loader is that all libraries should be loaded, and any weak symbols in those libraries should be ignored for normal symbols in any other library. This was indeed how weak symbol handling was originally implemented in Linux by glibc.
> 
> However, this was actually incorrect to the letter of the Unix standard at the time (SysVr4). The standard actually dictates that weak symbols should only be handled by the static linker; they should remain irrelevant to the dynamic linker (see the section on binding order below).

So I guess the right solution will be to remember all weak symbols and the corresponding library, and load them when no other definition was found.. But maybe just loading them is fine as long as it doesn't affect the startup time.